### PR TITLE
Make it possible to use type "area" for area cards instead of Mushroom template card

### PR DESCRIPTION
**Pull request by punxaphil**
## What
This provides an option to use `type: area` instead of mushroom-template-card for Areas, by configuring `use_ha_area_card: true`.
Along with that an option is also added to set common options for all area cards. Example: 
```yaml
strategy:
  type: custom:mushroom-strategy
  options:
    areas:
      _:
        use_ha_area_card: true
```
Of course it was already possible to override type in mushroom-strategy, but that wouldn't work with the `type: area`, since it also requires `area: <area id>`.

## Why
* The built in area card in Home Assistant supports pictures, which means that the user of mushroom-strategy now does not have to provide picture URLs for every area.
* The built in area-card shows pictures on the entire card, making it easier to see what is on the image (compared to the mushroom template card which only shows the picture as a round icon).

## Example
Example of how it can look:
image

### CODE REVIEW

**Comment by punxaphil** [comment]
@AalianKhan @ajagnanan @DigiLive  would love to hear your input on this 😄

**Comment by ajagnanan** [comment]
I think it's a good addition! Is there a way to move the section header `Areas` to align in the right column with the areas? That's happening on my dashboards as well.

Edit: I see it's fixed with https://github.com/AalianKhan/mushroom-strategy/pull/50

**Comment by DigiLive** [comment]
@johanfrick Thank you for your effort in enhancing the strategy.

I do like the looks, but I have some objections also...

1. It's called Mushroom Strategy, designed to auto-populate the dashboard with Mushroom cards, whenever useful and possible.
   I'm not sure we should mix all kind of card-styles together, but to keep it at a minimum instead.

2. I.m.o., If adding support for the [HA Area Card](https://www.home-assistant.io/dashboards/area/), the way to go is to add a new class to the package.
   The sole purpose of the current `AreaCard` class is to create a Mushroom-styled area card.

3. Adding properties to the configuration is allowed, so the following example should work.
   (However, additional properties may be incompatible with Mushroom Strategy.)

```yaml
    areas:
      biljardrum:
        type: area
        area: biljardrum
        navigation_path: biljardrum
```

My advice to @AalianKhan is to enable the [discussions](https://docs.github.com/discussions) in the settings of this repository and talk about this request and/or poll the user's opinions in a dedicated topic.

**Comment by punxaphil** [comment]
Yeah, you make a fair point that this is a strategy for Mushroom cards. However, this is also the greatest strategy I've seen for Home Assistant and I think the crowd can be made a lot wider if we allow some more customisation. Many people prefer Mushroom cards and nothing else, but not everyone does.
That being said, I'll give it another think how I can "hide" the intention of not using Mushroom card for some parts. Probably by moving more to the user's configuration of mushroom-strategy.
But let me know if it is not worth the effort, and I'll move forward by only maintaining my own fork.

**Comment by DigiLive** [comment]
Locked to continue the discussion at #58

File: README.md [hunk 1]
```diff
@@ -199,6 +199,33 @@ This area is enabled by default and includes the entities that aren't linked to
 The area can be configured like any other area as described above.
 To exclude this area from the dashboard and views, set its property `hidden` to `true`.
 
+#### Settings for all areas
+
+Use `_`to set options for all areas. Example:
```
Comment (DigiLive) [inline README.md]: Maybe `*` is a more appropriate wildcard indicator than `_`?

File: src/cards/AreaCard.js [hunk 2]
```diff
@@ -38,7 +38,12 @@ class AreaCard extends AbstractCard {
    */
   constructor(area, options = {}) {
     super(area);
-    this.#defaultOptions.primary                    = area.name;
+    for (const [key, value] of Object.entries(options)) {
```
Comment (DigiLive) [inline src/cards/AreaCard.js]: Following the "Single Responsibility Principle", this class's sole purpose is to create an Mushroom Area card.

Create a new class for creating an HA area card and instantiate the correct class in `HomeView.js`
This will make your current changes obsolete.

File: src/views/HomeView.js [hunk 3]
```diff
@@ -188,8 +188,11 @@ class HomeView extends AbstractView {
 
       for (const area of Helper.areas) {
         if (!Helper.strategyOptions.areas[area.area_id]?.hidden) {
+          let options = Helper.strategyOptions.areas[area.area_id ?? "undisclosed"];
```
Comment (DigiLive) [inline src/views/HomeView.js]: Import the correct module at line #186, depending on the type of `Helper.strategyOptions.areas["*"]`.

File: src/views/HomeView.js [hunk 4]
```diff
@@ -188,8 +188,11 @@ class HomeView extends AbstractView {
...
+          const optionsForAllAreas = Helper.strategyOptions.areas["_"];
+          options = {...optionsForAllAreas,...options}
```
Comment (DigiLive) [inline src/views/HomeView.js]: Merge the options in a single statement

```js
options = {
  ...Helper.strategyOptions.areas["*"];
  ...Helper.strategyOptions.areas[area.area_id ?? "undisclosed"],
}

File: README.md [hunk 5]
```diff
@@ -199,6 +199,33 @@ This area is enabled by default and includes the entities that aren't linked to
...
+  type: custom:mushroom-strategy
+  options:
+    areas:
+      _:
+        layout: vertical
+```
+
+##### Access to area ID
+
+If you want access to the area ID, you can use the value `"{{area_id}}"`. Example:
```
Comment (DigiLive) [inline README.md]: No need for this kind of evaluation if you create a new class for an HA Area card.

**Review by DigiLive — CHANGES_REQUESTED** [review]

**Comment by DigiLive** [comment]
I've unlocked this PR so the commits can be discussed.
Please discuss the more generic things in the discussion: #58

File: src/views/HomeView.js (hunk 4 above)
Comment (punxaphil) [reply src/views/HomeView.js]: I agree that * would be even nicer. The problem is that it is not supported by YAML, in the simple form. So I thought underscore would be the second best option. It is not very uncommon either to use underscore as wildcard in code. Examples:
* https://stackoverflow.com/questions/18300654/underscore-arrow-what-is-this
* https://stackoverflow.com/questions/41085189/using-underscore-variable-with-arrow-functions-in-es6-typescript

If you still don't like the underscore I'll adapt of course. Some other options I have been thinking about is:
```yaml
strategy:
  type: custom:mushroom-strategy
  options:
    areas:
      "*":
        type: area
        area: "{{area_id}}"
        navigation_path: "{{area_id}}"
```
```yaml
strategy:
  type: custom:mushroom-strategy
  options:
    all_areas:
      type: area
      area: "{{area_id}}"
      navigation_path: "{{area_id}}"
    areas:
      ...
```
```yaml
strategy:
  type: custom:mushroom-strategy
  options:
    areas:
      common_area_card_options:
        type: area
        area: "{{area_id}}"
        navigation_path: "{{area_id}}"
```

File: src/cards/AreaCard.js (hunk 2 above)
Comment (punxaphil) [reply src/cards/AreaCard.js]: This should be fixed now

File: src/views/HomeView.js (hunk 3 above)
Comment (punxaphil) [reply src/views/HomeView.js]: Done

File: README.md (hunk 5 above)
Comment (punxaphil) [reply README.md]: Done

**Comment by punxaphil** [comment]
Please review again @DigiLive . I have updated the code according to your comments. I have also updated the README, and the PR description.

File: src/views/HomeView.js (hunk 3 above)
Comment (DigiLive) [reply src/views/HomeView.js]: Actually, you didn't.

Instead of importing the correct module at line #186, you import both modules at line #3 and #4.
I requested something like:

```js
import("../cards/" + (Helper.strategyOptions.areas["_"]?.type ?? "AreaCard")).then(areaModule => {
```

You keep the module name `HaAreaCard`, but rename the class from `HaAreaCard` to `AreaCard`.
This way only 1 module will be imported (instead of 2) and we avoid an extra check for which class to instantiate.

File: src/typedefs.js [hunk 6]
```diff
@@ -118,6 +118,7 @@
  * @property {Object[]} [quick_access_cards] List of cards to show between welcome card and rooms cards.
  * @property {Object[]} [extra_cards] List of cards to show below room cards.
  * @property {Object[]} [extra_views] List of views to add to the dashboard.
+ * @property {boolean} [use_ha_area_card] Set to true to use ha area card instead of mushroom.
```
Comment (DigiLive) [inline src/typedefs.js]: Why did you move away from the proposed areas["_"] property?
Did I miss something?

File: src/cards/HaAreaCard.js [hunk 7]
```diff
@@ -0,0 +1,35 @@
...
+
+/**
+ * HA Area Card Class
+ *
+ * Used to create a card for an entity of the area domain using the built in type 'area'.
+ *
+ * @class
+ * @extends AbstractCard
+ */
+class HaAreaCard extends AbstractCard {
```
Comment (DigiLive) [inline src/cards/HaAreaCard.js]: Since the file is another module, you can give the class the same name `AreaCard` to avoid an extra check for which class to instantiate.

**Review by DigiLive — DISMISSED** [review]

File: src/typedefs.js (hunk 6 above)
Comment (punxaphil) [reply src/typedefs.js]: I guess I misunderstood you. I'll go back to "_" then :-)

File: src/cards/HaAreaCard.js (hunk 7 above)
Comment (punxaphil) [reply src/cards/HaAreaCard.js]: Ok!

File: src/views/HomeView.js (hunk 3 above)
Comment (punxaphil) [reply src/views/HomeView.js]: Ok! I'll change it

**Comment by punxaphil** [comment]
@DigiLive Ready for another review. I have updated according to your comments.

**Comment by DigiLive** [comment]
Please don't push any other commits as I'm reviewing and tweaking locally now.

I really want to avoid merge conflicts at this point.

**Comment by DigiLive** [comment]
@johanfrick Can you evaluate my code changes, please?

**Comment by punxaphil** [comment]
@DigiLive I have evaluated the code. It looks reasonable to me. I have also verified it by trying it out locally. I am happy to go forward with this solution. 👍

**Comment by punxaphil** [comment]
Anything else needed before merging?

**Review by AalianKhan — APPROVED** [review]

**Comment by maslyankov** [comment]
This looks well if the rooms have no temp/humidity sensors. If there are sensors for an area, they show up in the area card and therefore the UI looks too narrow and bad. The easiest solution would be to make them appear in one column.
